src/solidity/typeDecoder.js

```diff
@@ -0,0 +1,288 @@
+'use strict'
+
+/**
+  * determine if the given @arg type need to start from a new slot
+  *
+  * @param {Object} type - current type (object returned by the getType function)
+  * @return true if the type needs a new slot
+  */
+function typeNeedNewSlot (type) {
+  return (type.storageBytes === 'dynamic' || type.arraySize || type.members || type.fullType === 'string' || type.fullType === 'bytes')
+}
+
+/**
+  * parse the full type
+  *
+  * @param {String} fullType - type given by the AST (ex: uint[2] storage ref[2])
+  * @return {String} return the first part of the full type. keep the array declaration (uint[2][2] in that case)
```

Comment:
Please explain the distinction between full type and first part of the full type.